app/services/oauth_clients/set_attributes_service.rb

```diff
@@ -28,4 +28,19 @@
 
 # See also: create_service.rb for comments
 class OAuthClients::SetAttributesService < BaseServices::SetAttributes
+  private
+
+  def set_attributes(params)
+    super(replace_client_secret_with_nil(params))
+  end
+
+  def replace_client_secret_with_nil(params)
```

Comment:
Could you please clarify for me why we have to replace client secret with nil here? And can we remove it from params without clonning?

Reply:
Two answers:
- we replace the value of `""` with `nil`, because otherwise we get `""` in the database as a value, instead of `null`.
- we clone the params, because we want to maintain the immutability of method params.